src/Form/AdminSettingsForm.php

```diff
@@ -64,9 +64,13 @@ public function __construct(EntityManagerInterface $entity_manager, MessengerInt
    * {@inheritdoc}
    */
   public static function create(ContainerInterface $container) {
+    // MessengerInterface was introduced by Drupal 8.5.
+    // This code is for backwards-compatibility to 8.4 and below.
+    $messenger = $container->has('messenger') ? $container->get('messenger') : null;
+
     return new static(
       $container->get('entity.manager'),
```

Comment:
We should try not to use EntityManager either as it is deprecated and will be removed before D9.0, which is somewhere in the 8.x lifecycle.

Reply:
I agree the work needs to be done. I ran out of bandwidth so I'd recommend that we implement the update in a later iteration of update.